ground/src/main/java/com/google/android/ground/ui/map/Feature.kt

```diff
@@ -38,7 +45,14 @@ data class Feature(val tag: Tag, val geometry: Geometry) {
   enum class Type {
     /** This feature represents an unknown kind of object. */
     UNKNOWN,
-    /** This feature represents a single, discreet location on the map. */
+    /**
```

Comment:
Does `Feature` need to have a reference to the domain model at all? (e.g., location vs user polygon)? Can we abstract this to main better SoC between the map layer and business logic?

Reply:
@scolsen Gentle ping!

Reply:
`type` could be an `Int` or a `String` which would only have meaning for the caller. Alternatively, callers could prepend a namespace to the string `id`, though this is less desirable. Since the map layer doesn't know about LOIs or temporary map features for drawing tools, it would be up to the business logic and calling UI logic to set and interpret these fields. Does that make sense?

Reply:
Gotcha, yes I that makes sense -- so we'll just set it as `Int` and let callers decided the interpretation?

Only slight downside I see to that is that it does allow us to accidentally write code that misinterprets Features. For example, we could unintentionally pass a Feature representing a user-drawn polygon to LOI processing code and re-interpret it as an LOI if they happen to use the same convention (e.g. interpret 0 as my model type).

On the plus side this would give us much better separation between the layers.

Reply:
If can assume there will only be a single enum on the callers side with int values, the values will be compile checked. If we wanted we could provide the enum class as a generic to the map adapter, but this is likely overkill.

Reply:
Ahh, I see what you mean now -- so we'd have a single model-layer enum for distinguishing these. sgtm

Reply:
@gino-m is it closer to what you were thinking now?